internal/resource/configmap.go

```diff
@@ -129,3 +136,11 @@ func updateProperty(configMapData map[string]string, key string, value string) {
 		configMapData[key] = value
 	}
 }
+
+func removeHeadroom(memLimit int64) int64 {
+	const GiB int64 = 1073741824
+	if memLimit/5 > 2*GiB {
+		return memLimit - 2*GiB
+	}
+	return memLimit - memLimit/5
```

Comment:
Another neat pick: perhaps add a comment here stating that headroom is lowest between 20% of the amount passed in as argument or 2GiB?
I had to look at this for a bit to figure out why we were dividing by 5 😄

Reply:
Have added a comment to clarify why this exists and what it does

Reply:
Late to the party, but this "smaller amount of 20% memory or 2GiB" headroom should be documented, it was driving me crazy where my precious 20% of memory went :)

Reply:
It is documented:
https://www.rabbitmq.com/kubernetes/operator/using-operator#resource-reqs

Reply:
Ah I see now, couldn't find it, I was searching in the memory settings of rabbitmq docs but I also used the search bar to search for "20%" but that also didn't find this. Thanks you!